app/src/main/java/com/my/journeyplanner/views/itineraryresults/ItineraryResultsListAdapter.kt

```diff
@@ -0,0 +1,47 @@
+package com.my.journeyplanner.views.itineraryresults
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.my.core.domain.JourneyPlannerResultDomainModel.Itinerary.Journey
+import com.my.journeyplanner.R
+
+class ItineraryResultsListAdapter(private val itemClickListener: (List<Journey.Leg>) -> Unit) :
+    ListAdapter<Journey, ItineraryResultsListAdapter.MyViewHolder>(JourneyDiffCallback()) {
+    class MyViewHolder(var view: View) : RecyclerView.ViewHolder(view)
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
+        MyViewHolder(
+            LayoutInflater.from(parent.context)
+                .inflate(R.layout.recycler_text_views_itinerary_results, parent, false)
+        )
+
+    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
+        val view = holder.view
+        val context = view.context
+        val journey = getItem(position)
+        view.findViewById<TextView>(R.id.textViewStartTime).text = context.getString(
```

Comment:
because BindViewHolder is called multiple times for one view holder, it is wasteful to keep finding the views. The best approach is to use vals for these views in the ViewHolder class so you only find them once during the onCreateViewHolder fase.

Reply:
Done

Reply:
can't see it, have you comitted?

Reply:
Fixed